komga/src/test/kotlin/org/gotson/komga/domain/service/BookLifecycleTest.kt

```diff
@@ -130,4 +139,136 @@ class BookLifecycleTest(
     // then
     assertThat(readProgressRepository.findAll()).hasSize(2)
   }
+
+  @Test
+  fun `given a book with a sidecar when deleting book then all book files should be deleted`() {
+    Jimfs.newFileSystem(Configuration.unix()).use { fs ->
+      // given
+      val root = fs.getPath("/root")
+      Files.createDirectory(root)
+      val seriesPath = root.resolve("series")
+      Files.createDirectory(seriesPath)
+      val bookPath = seriesPath.resolve("book1.cbz")
+      Files.createFile(bookPath)
+      val sidecarPath = seriesPath.resolve("sidecar1.png")
+      Files.createFile(sidecarPath)
+
+      val series = makeSeries(name = "series", libraryId = library.id, url = seriesPath.toUri().toURL())
+      val book = makeBook("1", libraryId = library.id, url = bookPath.toUri().toURL())
+      val sidecar = ThumbnailBook(bookId = book.id, type = ThumbnailBook.Type.SIDECAR, url = sidecarPath.toUri().toURL())
+
+      seriesLifecycle.createSeries(series)
+      seriesLifecycle.addBooks(series, listOf(book))
+      thumbnailBookRepository.insert(sidecar)
+
+      // when
+      bookLifecycle.deleteBookFiles(book)
+
+      // then
+      assertThat(Files.notExists(bookPath))
+      assertThat(Files.notExists(sidecarPath))
+    }
+  }
+
+  @Test
+  fun `given a non-existent book file when deleting book then exception is thrown`() {
+    // given
+    val bookPath = Paths.get("/non-existent")
+    val book = makeBook("1", libraryId = library.id, url = bookPath.toUri().toURL())
+
+    // when
+    val thrown = catchThrowable { bookLifecycle.deleteBookFiles(book) }
+
+    // then
+    assertThat(thrown).hasCauseInstanceOf(FileNotFoundException::class.java)
+  }
+
+  @Test
+  fun `given a book and a non-existent sidecar file when deleting book then exception is thrown and no files are deleted`() {
```

Comment:
as mentioned in the service class, i think this is not a good behaviour, book should be deleted anyhow